usr/src/cmd/rcm_daemon/common/rcm_impl.h

```diff
@@ -103,7 +103,7 @@ typedef struct rcm_queue {
 
 #define	RCM_STRUCT_BASE_ADDR(struct_type, x, y)		\
 	((struct_type *) ((void *)(((char *)(x)) -	\
-			(int)(&((struct_type *)0)->y))))
+			(intptr_t)(&((struct_type *)0)->y))))
```

Comment:
This makes my brain hurt.

The second line, where you added the cast, is `offsetof` which would be better to just use.  The first line _might_ make this an equivalent to `containerof` (see `sys/containerof.h`.

Using `offsetof` is definitely better, if the whole things devolves into `containerof` just using `containerof` is better, but I'm not certain it does.

Reply:
The second part indeed translates into offsetof(struct_type, y)))) reading the definition of containerof() indeed is the
same when you replace the second part with offsetof().

#define RCM_STRUCT_BASE_ADDR(struct_type, x, y) \
__containerof(x, struct_type, y)